ca/ca.go

```diff
@@ -528,6 +531,51 @@ func (ca *CA) getTLSConfig(auth *authority.Authority) (*tls.Config, error) {
 	return tlsConfig, nil
 }
 
+// getTLSClientConfig returns a TLSConfig for the CA with a self-renewing client
+// certificate.
+func (ca *CA) getTLSClientConfig(auth *authority.Authority) (*tls.Config, error) {
+	// Create initial TLS certificate
+	tlsCrt, err := auth.GetTLSClientCertificate()
+	if err != nil {
+		return nil, err
+	}
+
+	// Start tls renewer with the new certificate.
+	// If a renewer was started, attempt to stop it before.
+	if ca.clientRenewer != nil {
+		ca.clientRenewer.Stop()
+	}
+
+	ca.clientRenewer, err = NewTLSRenewer(tlsCrt, auth.GetTLSClientCertificate)
+	if err != nil {
+		return nil, err
+	}
+	ca.renewer.Run()
+
+	var tlsConfig *tls.Config
+	if ca.config.TLS != nil {
+		tlsConfig = ca.config.TLS.TLSConfig()
+	} else {
+		tlsConfig = &tls.Config{
+			MinVersion: tls.VersionTLS12,
+		}
+	}
+
+	tlsConfig.GetClientCertificate = ca.clientRenewer.GetClientCertificateForCA
+
+	// Web PKI plus this authority's roots
+	certPool, err := x509.SystemCertPool()
```

Comment:
In design discussion we said only web PKI. Should I leave off the authority root?

Reply:
Authority root + WebPKI

Reply:
627ee201aaef760bbcf7a1ff26daf6caae9ed34a